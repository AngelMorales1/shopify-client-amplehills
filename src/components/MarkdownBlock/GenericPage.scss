@import 'settings', 'typography';

.MarkdownBlock {
  @extend .text-madison-blue;

  &--white {
    @extend .bg-white;
  }

  &--yellow {
    @extend .bg-marigold-yellow;
  }

  &--light-yellow {
    @extend .bg-off-yellow;
  }

  &--pink {
    @extend .bg-light-pink;
  }

  > h1 {
    @extend .title;
    margin-bottom: spacing(4);
  }

  > h2 {
    @extend .block-headline;
    margin-bottom: spacing(4);
  }

  > h3 {
    @extend .block-subheadline;
    @extend .bold;
    margin-bottom: spacing(2);
  }

  > p {
    @extend .block-subheadline;
    margin-bottom: spacing(8);

    > a {
      text-decoration: none;
      border-bottom: spacing(0.5) solid color('peach');
    }

    > strong {
      @extend .bold;
    }

    > img {
      width: 100%;
    }
  }

  ol,
  ul {
    margin: spacing(10);

    > li {
      margin: spacing(3) 0;
    }
  }

  blockquote {
    width: 100%;

    > p {
      width: 100%;
      text-align: center;
    }
  }
}

.MarkdownBlock--title-left {
  flex-direction: column;

  @media ($medium-up) {
    flex-direction: row;
  }

  .MarkdownBlock__title-container {
    padding: spacing(16) 0 0;

    @media ($medium-up) {
      padding: spacing(16) 0;
    }
  }
}
