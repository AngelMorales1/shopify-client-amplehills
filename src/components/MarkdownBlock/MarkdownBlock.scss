@import 'settings', 'typography', 'utils';

.MarkdownBlock {
  @extend .text-madison-blue;

  &--white {
    @extend .bg-white;
  }

  &--yellow {
    @extend .bg-marigold-yellow;
  }

  &--light-yellow {
    @extend .bg-off-yellow;
  }

  &--pink {
    @extend .bg-light-pink;
  }

  > h1 {
    @extend .title;
    margin-bottom: spacing(4);
  }

  > h2 {
    @extend .block-headline;
    margin-bottom: spacing(4);
  }

  > h3 {
    @extend .block-subheadline;
    @extend .bold;
    margin-bottom: spacing(2);
  }

  > p {
    @extend .block-subheadline;
    margin-bottom: spacing(8);

    > a {
      text-decoration: none;
      border-bottom: spacing(0.5) solid color('peach');
    }

    > strong {
      @extend .bold;
    }

    > img {
      width: 100%;
    }
  }

  ol,
  ul {
    margin: spacing(10);

    > li {
      margin: spacing(2) 0;
    }
  }

  ul {
    > li {
      position: relative;
    }
    > li::before {
      @extend .circle;

      background-color: color('peach');
      content: ' ';
      display: block;
      position: absolute;
      left: spacing(-6);
      top: calc(50% - #{6px});
      margin-bottom: 2px;
      width: 6px;
      height: 6px;
    }
  }

  blockquote {
    width: 100%;

    > p {
      width: 100%;
      text-align: center;
    }
  }
}

.MarkdownBlock--title-left {
  flex-direction: column;

  @media ($medium-up) {
    flex-direction: row;
  }

  .MarkdownBlock__title-container {
    padding: spacing(16) 0 0;

    @media ($medium-up) {
      padding: spacing(16) 0;
    }
  }
}
