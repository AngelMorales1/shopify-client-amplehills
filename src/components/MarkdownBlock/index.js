import React from 'react';
import marked from 'marked';

import cx from 'classnames';

import styles from './GenericPage.scss';

const MarkdownBlock = ({ content }) => {
  return (
    <div
      dangerouslySetInnerHTML={{ __html: marked(content) }}
      className={cx(
        styles['MarkdownBlock'],
        'form-container-width transition-slide-up mx-auto px3 py4'
      )}
    />
  );
};

export default MarkdownBlock;
