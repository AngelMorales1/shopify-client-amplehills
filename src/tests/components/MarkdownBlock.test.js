import React from 'react';
import { shallow } from 'enzyme';
import MarkdownBlock from 'components/MarkdownBlock';
it('renders without data', () => {
  const component = shallow(<MarkdownBlock />);
  expect(component).toMatchSnapshot();
});
it('renders with data', () => {
  const component = shallow(
    <MarkdownBlock
      z={0}
      block={{
        fields: {
          backgroundColor: 'foo',
          content: 'foo',
          drip: false,
          title: 'foo',
          titleLeft: true
        },
        sys: {
          contentType: {
            sys: {}
          },
          createdAt: 'foo',
          environment: {
            sys: {}
          },
          id: 'foo',
          locale: 'en-US',
          revision: 3,
          space: {
            sys: {}
          },
          type: 'foo',
          updatedAt: 'foo'
        }
      }}
    />
  );
  expect(component).toMatchSnapshot();
});
