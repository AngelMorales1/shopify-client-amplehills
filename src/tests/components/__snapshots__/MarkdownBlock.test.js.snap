// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders with data 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MarkdownBlock
    block={
        Object {
            "fields": Object {
              "backgroundColor": "foo",
              "content": "foo",
              "drip": false,
              "title": "foo",
              "titleLeft": true,
            },
            "sys": Object {
              "contentType": Object {
                "sys": Object {},
              },
              "createdAt": "foo",
              "environment": Object {
                "sys": Object {},
              },
              "id": "foo",
              "locale": "en-US",
              "revision": 3,
              "space": Object {
                "sys": Object {},
              },
              "type": "foo",
              "updatedAt": "foo",
            },
          }
    }
    setRef={[Function]}
    z={0}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="w100 flex content-width undefined"
>
        <div
                className="col-12 md-col-4"
        >
                <h2
                        className="block-headline"
                >
                        foo
                </h2>
        </div>
        <div
                className="transition-slide-up mx-auto py4 col-12 md-col-8"
        >
                <div
                        className="col-12 markdown-block form-container-width"
                        dangerouslySetInnerHTML={
                                Object {
                                        "__html": "<p>foo</p>
                                ",
                                      }
                        }
                />
        </div>
</div>,
      "className": "flex justify-center px3",
      "style": Object {
        "zIndex": 0,
      },
    },
    "ref": [Function],
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="col-12 md-col-4"
>
            <h2
                        className="block-headline"
            >
                        foo
            </h2>
</div>,
          <div
            className="transition-slide-up mx-auto py4 col-12 md-col-8"
>
            <div
                        className="col-12 markdown-block form-container-width"
                        dangerouslySetInnerHTML={
                                    Object {
                                                "__html": "<p>foo</p>
                                    ",
                                              }
                        }
            />
</div>,
        ],
        "className": "w100 flex content-width undefined",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <h2
              className="block-headline"
>
              foo
</h2>,
            "className": "col-12 md-col-4",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "foo",
              "className": "block-headline",
            },
            "ref": null,
            "rendered": "foo",
            "type": "h2",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="col-12 markdown-block form-container-width"
                dangerouslySetInnerHTML={
                                Object {
                                                "__html": "<p>foo</p>
                                ",
                                              }
                }
/>,
              null,
            ],
            "className": "transition-slide-up mx-auto py4 col-12 md-col-8",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "col-12 markdown-block form-container-width",
                "dangerouslySetInnerHTML": Object {
                  "__html": "<p>foo</p>
",
                },
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
            null,
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="w100 flex content-width undefined"
>
          <div
                    className="col-12 md-col-4"
          >
                    <h2
                              className="block-headline"
                    >
                              foo
                    </h2>
          </div>
          <div
                    className="transition-slide-up mx-auto py4 col-12 md-col-8"
          >
                    <div
                              className="col-12 markdown-block form-container-width"
                              dangerouslySetInnerHTML={
                                        Object {
                                                  "__html": "<p>foo</p>
                                        ",
                                                }
                              }
                    />
          </div>
</div>,
        "className": "flex justify-center px3",
        "style": Object {
          "zIndex": 0,
        },
      },
      "ref": [Function],
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="col-12 md-col-4"
>
              <h2
                            className="block-headline"
              >
                            foo
              </h2>
</div>,
            <div
              className="transition-slide-up mx-auto py4 col-12 md-col-8"
>
              <div
                            className="col-12 markdown-block form-container-width"
                            dangerouslySetInnerHTML={
                                          Object {
                                                        "__html": "<p>foo</p>
                                          ",
                                                      }
                            }
              />
</div>,
          ],
          "className": "w100 flex content-width undefined",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <h2
                className="block-headline"
>
                foo
</h2>,
              "className": "col-12 md-col-4",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "foo",
                "className": "block-headline",
              },
              "ref": null,
              "rendered": "foo",
              "type": "h2",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="col-12 markdown-block form-container-width"
                  dangerouslySetInnerHTML={
                                    Object {
                                                      "__html": "<p>foo</p>
                                    ",
                                                    }
                  }
/>,
                null,
              ],
              "className": "transition-slide-up mx-auto py4 col-12 md-col-8",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "col-12 markdown-block form-container-width",
                  "dangerouslySetInnerHTML": Object {
                    "__html": "<p>foo</p>
",
                  },
                },
                "ref": null,
                "rendered": null,
                "type": "div",
              },
              null,
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`renders without data 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MarkdownBlock
    block={
        Object {
            "fields": Object {
              "backgroudColor": "white",
              "content": "",
              "drip": false,
              "title": "",
              "titleLeft": false,
            },
          }
    }
    setRef={[Function]}
    z={0}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="w100 flex content-width"
>
        <div
                className="transition-slide-up mx-auto py4"
        >
                <div
                        className="col-12 markdown-block form-container-width"
                        dangerouslySetInnerHTML={
                                Object {
                                        "__html": "",
                                      }
                        }
                />
        </div>
</div>,
      "className": "flex justify-center px3",
      "style": Object {
        "zIndex": 0,
      },
    },
    "ref": [Function],
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          null,
          <div
            className="transition-slide-up mx-auto py4"
>
            <div
                        className="col-12 markdown-block form-container-width"
                        dangerouslySetInnerHTML={
                                    Object {
                                                "__html": "",
                                              }
                        }
            />
</div>,
        ],
        "className": "w100 flex content-width",
      },
      "ref": null,
      "rendered": Array [
        null,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="col-12 markdown-block form-container-width"
                dangerouslySetInnerHTML={
                                Object {
                                                "__html": "",
                                              }
                }
/>,
              null,
            ],
            "className": "transition-slide-up mx-auto py4",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "col-12 markdown-block form-container-width",
                "dangerouslySetInnerHTML": Object {
                  "__html": "",
                },
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
            null,
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="w100 flex content-width"
>
          <div
                    className="transition-slide-up mx-auto py4"
          >
                    <div
                              className="col-12 markdown-block form-container-width"
                              dangerouslySetInnerHTML={
                                        Object {
                                                  "__html": "",
                                                }
                              }
                    />
          </div>
</div>,
        "className": "flex justify-center px3",
        "style": Object {
          "zIndex": 0,
        },
      },
      "ref": [Function],
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            null,
            <div
              className="transition-slide-up mx-auto py4"
>
              <div
                            className="col-12 markdown-block form-container-width"
                            dangerouslySetInnerHTML={
                                          Object {
                                                        "__html": "",
                                                      }
                            }
              />
</div>,
          ],
          "className": "w100 flex content-width",
        },
        "ref": null,
        "rendered": Array [
          null,
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="col-12 markdown-block form-container-width"
                  dangerouslySetInnerHTML={
                                    Object {
                                                      "__html": "",
                                                    }
                  }
/>,
                null,
              ],
              "className": "transition-slide-up mx-auto py4",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "col-12 markdown-block form-container-width",
                  "dangerouslySetInnerHTML": Object {
                    "__html": "",
                  },
                },
                "ref": null,
                "rendered": null,
                "type": "div",
              },
              null,
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
